src/Tools/dotnet-monitor/CollectionRules/Configuration/CollectionRulePostConfigureNamedOptions.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Globalization;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Configuration
+{
+    internal sealed class CollectionRulePostConfigureOptions :
+        IPostConfigureOptions<CollectionRuleOptions>
+    {
+        private readonly ICollectionRuleActionOperations _actionOperations;
+        private readonly ICollectionRuleTriggerOperations _triggerOperations;
+        private readonly TemplateOptions _templateOptions;
+        private readonly IConfiguration _configuration;
+
+        public CollectionRulePostConfigureOptions(
+            IOptionsMonitor<TemplateOptions> templateOptions,
+            IConfiguration configuration,
+            ICollectionRuleActionOperations actionOperations,
+            ICollectionRuleTriggerOperations triggerOperations)
+        {
+            _templateOptions = templateOptions.CurrentValue;
+            _configuration = configuration;
+            _actionOperations = actionOperations;
+            _triggerOperations = triggerOperations;
+        }
+
+        public void PostConfigure(string name, CollectionRuleOptions options)
+        {
+            IConfigurationSection ruleSection = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name));
+
+            ResolveActionList(options, ruleSection);
+            ResolveTrigger(options, ruleSection);
+            ResolveFilterList(options, ruleSection);
+            ResolveLimits(options, ruleSection);
+        }
+
+        private void ResolveActionList(CollectionRuleOptions ruleOptions, IConfigurationSection ruleSection)
+        {
+            ruleOptions.Actions.Clear();
```

Comment:
We're clearing out the action list to avoid having to insert actions at the right spots - Justin wanted to avoid having to figure out the right index for where to put templates to maintain ordering, so we just re-populate the full set of actions from scrach here, regardless of whether or not they're templates.